models/aeif_cond_beta_multisynapse.cpp

```diff
@@ -168,60 +156,54 @@ aeif_cond_beta_multisynapse::Parameters_::set( const DictionaryDatum& d )
   updateValue< double >( d, names::C_m, C_m );
   updateValue< double >( d, names::g_L, g_L );
 
-  std::vector< double > Erev_tmp, taur_tmp, taud_tmp;
+  size_t old_n_receptors = n_receptors();
   bool Erev_flag =
-    updateValue< std::vector< double > >( d, names::E_rev, Erev_tmp );
+    updateValue< std::vector< double > >( d, names::E_rev, E_rev );
   bool taur_flag =
-    updateValue< std::vector< double > >( d, names::taus_rise, taur_tmp );
+    updateValue< std::vector< double > >( d, names::taus_rise, taus_rise );
   bool taud_flag =
-    updateValue< std::vector< double > >( d, names::taus_decay, taud_tmp );
+    updateValue< std::vector< double > >( d, names::taus_decay, taus_decay );
   if ( Erev_flag || taur_flag || taud_flag )
```

Comment:
I think it should be possible to simplify this some more:
```C++
// n_receptors() == E_rev.size()
if ( n_receptors() != taus_rise.size() || n_receptors() != taus_decay.size() )
{
  throw BadProperty("E_rev, taus_raus and taus_decays must have the same size.");
}
```
Now we know all is consistent I would then check `n_receptors() != old_n_receptors && has_connections` and `n_receptors > 0`. Then the check that the taus > 0 and tau_rise < tau_decay. That should maybe be protected by `if ( taur_flag || taud_flag )` since we have loops here and would want to avoid running through them if the taus have not been changed.